ShellPkg/Library/UefiShellBcfgCommandLib/UefiShellBcfgCommandLib.c

```diff
@@ -1586,6 +1626,12 @@ ShellCommandRunBcfg (
     if ((ShellStatus == SHELL_SUCCESS) && (CurrentOperation.Target < BcfgTargetMax)) {
       for (ParamNumber = 2; ParamNumber < ShellCommandLineGetCount (Package) && ShellStatus == SHELL_SUCCESS; ParamNumber++) {
         CurrentParam = ShellCommandLineGetRawValue (Package, ParamNumber);
+        if (CurrentParam == NULL) {
+          ShellPrintHiiEx (-1, -1, NULL, STRING_TOKEN (STR_GEN_PARAM_INV), gShellBcfgHiiHandle, L"bcfg", L"NULL");
+          ShellStatus = SHELL_INVALID_PARAMETER;
+          return (ShellStatus);

```

Comment:
Similar comment, it seems `return` statements are avoided in this function to fallback to:

```
  if (Package != NULL) {
    ShellCommandLineFreeVarList (Package);
  }

  if (CurrentOperation.FileName != NULL) {
    FreePool (CurrentOperation.FileName);
  }

  if (CurrentOperation.Description != NULL) {
    FreePool (CurrentOperation.Description);
  }

  if (CurrentOperation.Order != NULL) {
    FreePool (CurrentOperation.Order);
  }
```
in all cases

Reply:
Agreed. Added a goto here instead.